src/Components/FakeTargetsOutline.fs

```diff
@@ -0,0 +1,369 @@
+namespace Ionide.VSCode.FSharp
+
+open System
+open Fable.Core
+open Fable.Core.JsInterop
+open Fable.Import
+open Fable.Import.vscode
+open Fable.Import.Node
+open Ionide.VSCode.Helpers
+open System.Collections.Generic
+
+open DTO
+open DTO.FakeSupport
+open Ionide.VSCode.Helpers
+module node = Fable.Import.Node.Exports
+
+module FakeTargetsOutline =
+
+    let private configurationKey = "FAKE.targetsOutline"
+    let private isEnabledFor uri = configurationKey |> Configuration.getInContext uri true
+
+    type NodeEntry =
+        { Key : string
+          Children : Dictionary<string, NodeEntry>
+          Symbol : Symbol }
+
+    type DependencyType =
+        | SoftDependency
+        | HardDependency
+        member x.Arrow =
+            match x with
+            | SoftDependency -> "<=?"
+            | HardDependency -> "<=="
+    type ModelType =
+        | TargetModel
+        | DependencyModel of DependencyType
+
+    type Model =
+        { AllTargets : Target []
+          Label : string
+          Description : string
+          Declaration : Declaration
+          Type : ModelType
+          getChildren : unit -> ResizeArray<Model> }      
+
+    let refresh = EventEmitter<Uri> ()
+    let private reallyRefresh = EventEmitter<Model option> ()
+    let mutable private currentDocument : string option = None
+
+    let private getIconPath light dark =
+        let plugPath =
+            try
+                (VSCode.getPluginPath "Ionide.ionide-fsharp")
+            with
+            | _ ->  (VSCode.getPluginPath "Ionide.Ionide-fsharp")
+
+        let p = createEmpty<TreeIconPath>
+        p.dark <- node.path.join(plugPath, "images", dark) |> U3.Case1
+        p.light <- node.path.join(plugPath, "images", light) |> U3.Case1
+        p
+
+    let rec add' (state : NodeEntry) (symbol : Symbol) index =
+        let sep = "."
+
+        let entry = symbol.Name
+
+        if index >= entry.Length then
+            state
+        else
+            let endIndex = entry.IndexOf(sep, index)
+            let endIndex = if endIndex = -1 then entry.Length else endIndex
+
+            let key = entry.Substring(index, endIndex - index)
+            if String.IsNullOrEmpty key then
+                state
+            else
+                if state.Children.ContainsKey key |> not then
+                    let x = {Key = key; Children = new Dictionary<_,_>(); Symbol = symbol}
+                    state.Children.Add(key,x)
+                let item = state.Children.[key]
+                add' item symbol (endIndex + 1)
+
+    let getRoot (doc : TextDocument) =
+        promise {
+            let! o = LanguageService.FakeSupport.targetsInfo doc.fileName
```

Comment:
Initially, I thought it is a bad idea to just execute a fake script (with potential side-effects) when opening a script in Ionide.
But Ionide is doing something similar today: You can just hook into msbuild - like paket does - and there you go: arbitrary code execution.

So the only real reason against doing it that way is if we think people have hurting side-effects in their script by accident. But given the design of FAKE such scripts are probably not that useful (and therefore unlikely to exist).